qiskit/transpiler/passes/utils/check_gate_direction.py

```diff
@@ -33,6 +34,40 @@ def __init__(self, coupling_map, target=None):
         self.coupling_map = coupling_map
         self.target = target
 
+    def _coupling_map_visit(self, dag, wire_map, edges=None):
+        if edges is None:
+            edges = self.coupling_map.get_edges()
+        for node in dag.two_qubit_ops():
+            if isinstance(node.op, ControlFlowOp):
+                continue
+            if (wire_map[node.qargs[0]], wire_map[node.qargs[1]]) not in edges:
+                return False
+        for node in dag.op_nodes(ControlFlowOp):
+            for block in node.op.blocks:
+                inner_wire_map = {
+                    inner: wire_map[outer] for outer, inner in zip(node.qargs, block.qubits)
+                }
+                if not self._coupling_map_visit(circuit_to_dag(block), inner_wire_map, edges):
+                    return False
```

Comment:
I'm happy to do this.  We're already fairly inefficient when dealing with control-flow, because there a quite a few cases where we first iterate over all control-flow nodes, and then do whatever the normal pass handling is, and of course there's the _repeated_ calls to `circuit_to_dag` for the time being.

Reply:
Sure, I was more just worried that this adds a second iteration over all the nodes even if there is no control flow because `dag.op_nodes(ControlFlowOp)` internally loops over all the nodes to get the list of control flow ops even if it's empty.

Reply:
Done in eac245e48.